From 28aa315414541746566b257f20ee20d0b49a4c7d Mon Sep 17 00:00:00 2001
From: Dorian Stoll <dorian.stoll@tmsp.io>
Date: Thu, 25 Apr 2019 19:23:36 +0200
Subject: [PATCH] mkspec

Signed-off-by: Dorian Stoll <dorian.stoll@tmsp.io>
---
 scripts/package/mkspec | 67 ++++++++++++++++--------------------------
 1 file changed, 25 insertions(+), 42 deletions(-)

diff --git a/scripts/package/mkspec b/scripts/package/mkspec
index 009147d4718e..812116c04351 100755
--- a/scripts/package/mkspec
+++ b/scripts/package/mkspec
@@ -27,7 +27,6 @@ if grep -q CONFIG_DRM=y .config; then
 	PROVIDES=kernel-drm
 fi
 
-PROVIDES="$PROVIDES kernel-$KERNELRELEASE"
 __KERNELRELEASE=$(echo $KERNELRELEASE | sed -e "s/-/_/g")
 EXCLUDES="$RCS_TAR_IGNORE --exclude=.tmp_versions --exclude=*vmlinux* \
 --exclude=*.o --exclude=*.ko --exclude=*.cmd --exclude=Documentation \
@@ -39,42 +38,31 @@ EXCLUDES="$RCS_TAR_IGNORE --exclude=.tmp_versions --exclude=*vmlinux* \
 #  $S: this line is enabled only when building source package
 #  $M: this line is enabled only when CONFIG_MODULES is enabled
 sed -e '/^DEL/d' -e 's/^\t*//' <<EOF
-	Name: kernel
+	Name: kernel-surface
 	Summary: The Linux Kernel
-	Version: $__KERNELRELEASE
+	Version: $(echo $KERNELRELEASE | cut -d'-' -f1)
 	Release: $(cat .version 2>/dev/null || echo 1)
 	License: GPL
 	Group: System Environment/Kernel
 	Vendor: The Linux Community
 	URL: http://www.kernel.org
 $S	Source: kernel-$__KERNELRELEASE.tar.gz
-	Provides: $PROVIDES
+	Provides: kernel $PROVIDES
 	%define __spec_install_post /usr/lib/rpm/brp-compress || :
 	%define debug_package %{nil}
 
 	%description
 	The Linux Kernel, the operating system core itself
 
-	%package headers
-	Summary: Header files for the Linux kernel for use by glibc
-	Group: Development/System
-	Obsoletes: kernel-headers
-	Provides: kernel-headers = %{version}
-	%description headers
-	Kernel-headers includes the C header files that specify the interface
-	between the Linux kernel and userspace libraries and programs.  The
-	header files define structures and constants that are needed for
-	building most standard programs and are also needed for rebuilding the
-	glibc package.
-
-$S$M	%package devel
-$S$M	Summary: Development package for building kernel modules to match the $__KERNELRELEASE kernel
-$S$M	Group: System Environment/Kernel
-$S$M	AutoReqProv: no
-$S$M	%description -n kernel-devel
-$S$M	This package provides kernel headers and makefiles sufficient to build modules
-$S$M	against the $__KERNELRELEASE kernel package.
-$S$M
+$M	%package devel
+$M	Summary: Development package for building kernel modules to match the $__KERNELRELEASE kernel
+$M	Group: System Environment/Kernel
+$M	AutoReqProv: no
+$M	Provides: kernel-devel
+$M	%description -n kernel-surface-devel
+$M	This package provides kernel headers and makefiles sufficient to build modules
+$M	against the $__KERNELRELEASE kernel-surface package.
+$M
 $S	%prep
 $S	%setup -q
 $S
@@ -91,18 +79,17 @@ $S
 	cp \$($MAKE image_name) %{buildroot}/boot/vmlinuz-$KERNELRELEASE
 	%endif
 $M	$MAKE %{?_smp_mflags} INSTALL_MOD_PATH=%{buildroot} modules_install
-	$MAKE %{?_smp_mflags} INSTALL_HDR_PATH=%{buildroot}/usr headers_install
 	cp System.map %{buildroot}/boot/System.map-$KERNELRELEASE
 	cp .config %{buildroot}/boot/config-$KERNELRELEASE
 	bzip2 -9 --keep vmlinux
 	mv vmlinux.bz2 %{buildroot}/boot/vmlinux-$KERNELRELEASE.bz2
-$S$M	rm -f %{buildroot}/lib/modules/$KERNELRELEASE/build
-$S$M	rm -f %{buildroot}/lib/modules/$KERNELRELEASE/source
-$S$M	mkdir -p %{buildroot}/usr/src/kernels/$KERNELRELEASE
-$S$M	tar cf - $EXCLUDES . | tar xf - -C %{buildroot}/usr/src/kernels/$KERNELRELEASE
-$S$M	cd %{buildroot}/lib/modules/$KERNELRELEASE
-$S$M	ln -sf /usr/src/kernels/$KERNELRELEASE build
-$S$M	ln -sf /usr/src/kernels/$KERNELRELEASE source
+$M	rm -f %{buildroot}/lib/modules/$KERNELRELEASE/build
+$M	rm -f %{buildroot}/lib/modules/$KERNELRELEASE/source
+$M	mkdir -p %{buildroot}/usr/src/kernels/$KERNELRELEASE
+$M	tar cf - $EXCLUDES . | tar xf - -C %{buildroot}/usr/src/kernels/$KERNELRELEASE
+$M	cd %{buildroot}/lib/modules/$KERNELRELEASE
+$M	ln -sf /usr/src/kernels/$KERNELRELEASE build
+$M	ln -sf /usr/src/kernels/$KERNELRELEASE source
 
 	%clean
 	rm -rf %{buildroot}
@@ -134,14 +121,10 @@ $M	/lib/modules/$KERNELRELEASE
 $M	%exclude /lib/modules/$KERNELRELEASE/build
 $M	%exclude /lib/modules/$KERNELRELEASE/source
 	/boot/*
-
-	%files headers
-	%defattr (-, root, root)
-	/usr/include
-$S$M
-$S$M	%files devel
-$S$M	%defattr (-, root, root)
-$S$M	/usr/src/kernels/$KERNELRELEASE
-$S$M	/lib/modules/$KERNELRELEASE/build
-$S$M	/lib/modules/$KERNELRELEASE/source
+$M
+$M	%files devel
+$M	%defattr (-, root, root)
+$M	/usr/src/kernels/$KERNELRELEASE
+$M	/lib/modules/$KERNELRELEASE/build
+$M	/lib/modules/$KERNELRELEASE/source
 EOF
-- 
2.21.0

